source/fem-dem/vans_assemblers.cc

```diff
@@ -1820,7 +1824,9 @@ VANSAssemblerBuoyancy<dim>::calculate_particle_fluid_interactions(
     RequiresConstantDensity(
       "VANSAssemblerBuoyancy<dim>::calculate_particle_fluid_interactions"));
 
-  const double density = scratch_data.properties_manager.get_density_scale();
+  const std::vector<double> density = scratch_data.density_at_particle_location;
```

Comment:
```suggestion
  const std::vector<double>& density = scratch_data.density_at_particle_location;
```
This should be a reference not a copy. Otherwise you are copying the array for every time.